### DISCUSSION TITLE
Additional Optocoupler SH-ESP32

### DISCUSSION THREAD

**Discussion by mistljo** [hatlabs/discussions#25] [Q&A]
HI Matti,
I want to add additional input optocouplers on the SH-ESP32. For 12 volt boat system and for soldering directly to the board (no SMD). Can you recommend such a optocoupler that have the same specification as the existing one?
Thanks Joachim

**Comment by mairas** [comment]
Hi Joachim,

Seems like optocouplers are all the rage at the moment, someone else was just asking about them as well!

The SH-ESP32 optocouplers are these:

https://lcsc.com/product-detail/Optocouplers-Phototransistor-Output_Everlight-Elec-EL3H7-C-TA-G_C92243.html

They're generic "jellybean" components that are basically SMD-packaged copies of Sharp's PC817.

If you go to any electronics supplier and search for "817 optocoupler", you'll find loads of suitable components. If you want to add more than just one optocoupler, choose one that has multiple channels in the same package. In that same family, 847 are four-channel ones. For example, LTV-847, available at least on Mouser and LCSC, seems to be superficially very close. But double-check the data sheets to verify suitability before you order!

Cheers,

Matti
